backend/src/main/java/com/example/backend/service/AuthenticationService.java

```diff
@@ -127,6 +127,18 @@ public void disableAccountByUserId(String userId) {
         }
     }
 
+    public void unlockAccountByUserId(String userId) {
+        int rowsAffected = userDetailsService.unlockAppUser(userId);
+
+        if(rowsAffected == 0) {
+            throw new UserIdNotFoundException(userId);
+        }
+    }
+
+    public boolean isAccountLockedByEmail(String userId) {
```

Comment:
The String argument should be `email` not `userId` since the method `isAccountLocked` provided in class `AppUserDetailsService` it's using the email as argument.